packages/react-refresh/src/__tests__/ReactFreshBabelPlugin-test.js

```diff
@@ -7,25 +7,54 @@
 
 'use strict';
 
-let babel = require('babel-core');
+let babel = require('@babel/core');
 let {wrap} = require('jest-snapshot-serializer-raw');
 let freshPlugin = require('react-refresh/babel');
 
 function transform(input, options = {}) {
   return wrap(
     babel.transform(input, {
       babelrc: false,
+      configFile: false,
       plugins: [
-        'syntax-jsx',
-        'syntax-dynamic-import',
+        '@babel/syntax-jsx',
+        '@babel/syntax-dynamic-import',
         freshPlugin,
         ...(options.plugins || []),
       ],
     }).code,
   );
 }
 
+const oldEnv = babel.getEnv();
 describe('ReactFreshBabelPlugin', () => {
+  beforeEach(() => {
```

Comment:
The reason Fresh throws an error if you're not in development environment is because the production build of React does not support hot reloading. The production build of React uses a `process.env.NODE_ENV` check (that's what `__DEV__` compiles to), not `BABEL_ENV`. Technically, `BABEL_ENV` and `NODE_ENV` could be different, even though they're usually the same.

We should use `__DEV__`/`NODE_ENV` since that's what determines the version of React DOM.

```js
if (__DEV__) {
  // The tests
} else {
  it('throw error if environment is not development', ...);
}
```